cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1535,10 +1535,41 @@ public void setTrackingConsent(Boolean consent) {
       for (TrackingConsentListener listener : trackingConsentListeners) {
         listener.ready();
       }
+    } else {
+      removeSubscriptionTagsAndAttributes();
```

Comment:
Only call this when hasTrackingConsent was `true` before this method was called. So we need to compare the previous value before overriding it here https://github.com/cleverpush/cleverpush-android-sdk/pull/174/files#diff-dadb76999f5ed7d5f6c363d011ef4b354183a8b5cee4ecd118f2152c882891f1R1532